src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/CollectionSettings.java

```diff
@@ -106,4 +97,41 @@ public static CollectionSettings getCollectionSettings(
           null);
     }
   }
+
+  public static CollectionSettings getCollectionSettings(
+      String collectionName,
+      boolean vectorEnabled,
+      int vectorSize,
+      SimilarityFunction similarityFunction,
+      String vectorize,
+      ObjectMapper objectMapper) {
+    // parse vectorize to get vectorizeServiceName and modelName
+    if (vectorize != null && !vectorize.isBlank()) {
+      return createCollectionSettingsFromJson(
+          collectionName, vectorEnabled, vectorSize, similarityFunction, vectorize, objectMapper);
+    } else {
+      return new CollectionSettings(
+          collectionName, vectorEnabled, vectorSize, similarityFunction, null, null);
+    }
+  }
+
+  private static CollectionSettings createCollectionSettingsFromJson(
+      String collectionName,
+      boolean vectorEnabled,
+      int vectorSize,
+      SimilarityFunction function,
+      String vectorize,
+      ObjectMapper objectMapper) {
+    try {
+      JsonNode vectorizeConfig = objectMapper.readTree(vectorize);
+      String vectorizeServiceName = vectorizeConfig.get("service").textValue();
+      JsonNode optionsNode = vectorizeConfig.get("options");
+      String modelName = optionsNode.get("modelName").textValue();
+      return new CollectionSettings(
+          collectionName, vectorEnabled, vectorSize, function, vectorizeServiceName, modelName);
+    } catch (JsonProcessingException e) {
+      // This should never happen
```

Comment:
It would happen if String that is supposed to contain JSON was corrupted and isn't valid JSON, so should probably throw some more specific exception that indicates invalid JSON?